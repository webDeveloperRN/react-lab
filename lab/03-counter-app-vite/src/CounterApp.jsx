import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ( {value} ) => {

    const [ counter, setCounter ] = useState( value );

    const handleAdd = () => {
        setCounter( counter +  1);
        //setCounter( (c) => + 1):
    }
    const handleMinus = () =>  setCounter( counter -  1);
    const handleReset = () =>  setCounter( value ); 
    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>
            <button onClick={ handleAdd } >+1</button>
            <button onClick={ handleMinus } >-1</button>
            <button onClick={ handleReset } >Reset</button>
        </> 
    )
}

CounterApp.propTypes = {
    value : PropTypes.number.isRequired
}