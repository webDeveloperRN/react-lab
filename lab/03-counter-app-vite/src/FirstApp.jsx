import PropTypes from 'prop-types';

export const FirstApp = ({ title, subTitle, name }) => {
    return (
        //Estp es in fragment <></> es un wrapper o padre
        <>
            <h1>{ title }</h1>
            <p>{ subTitle }</p>
            <p>{ subTitle }</p>
        </>
    );
};
FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
  }
  
  FirstApp.defaultProps = {
    name: 'Fernando Herrera',
    subTitle: 'No hay subtítulo',
    // title: 'No hay título',
  }