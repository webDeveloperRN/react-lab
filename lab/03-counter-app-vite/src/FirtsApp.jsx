import PropTypes from 'prop-types';

export const FirtsApp = ( {title, subTitle, name} ) => {
    return (
        //Estp es in fragment <></> es un wrapper o padre
        <>
            <p>{ title }</p>
            <p>{ subTitle }</p>
            <p>{ name }</p>
            <p>Segundo nodeo</p>
        </>
    );
};

FirtsApp.propTypes = {
    title: PropTypes.string.isRequired
}

//Default props
FirtsApp.defaultProps = {
    title: 'No hay titulo',
    subTitle: 'No hay subtitulo',
    name: 'Fran'
}