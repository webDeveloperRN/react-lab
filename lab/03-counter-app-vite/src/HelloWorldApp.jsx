export const  HelloWorldApp = () => {
    return (
        <h1>Hello app</h1>
    );
} 