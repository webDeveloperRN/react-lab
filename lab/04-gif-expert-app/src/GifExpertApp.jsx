import { useState } from 'react';
import { AddCategory } from './components/AddCategory';


export const GifExpertApp = () => {
    const [categories, setcategories] = useState( ['One Punch', 'Dragon ball']);
    const onAddCategory = () => {
        //Forma  1
        //setcategories(categories => [...categories, 'Valorant']);
        //Forma2
        //setcategories( categories => [...categories, 'Valorant'] );
    }
    return(
        <>
        { /*titulo */ }
        <h1>GifExpertApp</h1>
        { /* input */ }
        <AddCategory setcategories={ setcategories } />
        { /* Listado de gifs */}
        <ol>
            {
            categories.map( category => {
                return <li key={ category }>{ category}</li>
            })
            }
        </ol>
        </>
    )
}
//hiD9ZAAgrs7PWYA3ZGNBva1oT8Kfch3l