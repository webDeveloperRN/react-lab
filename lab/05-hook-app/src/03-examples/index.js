export * from './LoadingQuotes';
export * from './Quote';
export * from './MultipleCustomHooks';