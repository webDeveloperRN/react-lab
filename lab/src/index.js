 

const promesa = new Promise( (resolve, reject) => {
    setTimeout( () => {
        console.log("2 segundos despues");
    }, 2000);
});

promesa.then( () => {
    console.log("Then de la promesa");
});