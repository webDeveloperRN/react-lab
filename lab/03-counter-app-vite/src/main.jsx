import React from "react";
//Heraminetaz de rdenrizado
import ReactDOM from "react-dom/client";
//nuestro funcitional component
//import { HelloWorldApp } from './HelloWorldApp';
import './styles.css'

//import  {FirtsApp} from './FirtsApp';
import  {CounterApp} from './CounterApp';
//Renderizado del compoenent
ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        <CounterApp value={ 0 }  />
    </React.StrictMode>
);