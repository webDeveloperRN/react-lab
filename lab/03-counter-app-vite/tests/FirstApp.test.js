/**
 * @jest-environment jsdom
 */
import { render } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';


describe('Pruebas en <FirstApp />', () => {
    
    test('debe de hacer match con el snapshot', () => {    
        const title = 'Hola, Soy Goku';
        const { container } = render( <FirstApp title={ title } /> );

        expect( container ).toMatchSnapshot();

     }); 
});