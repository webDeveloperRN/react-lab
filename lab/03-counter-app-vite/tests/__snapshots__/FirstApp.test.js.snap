// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <p>
    Hola, Soy Goku
  </p>
  <p>
    No hay subt√≠tulo
  </p>
  <p>
    Fernando Herrera
  </p>
</div>
`;
